### ISSUE TITLE
Deprecating the old importing mechanism

### ISSUE THREAD

**Issue by RiccardoRossi** [KratosMultiphysics/Kratos#5268]
the old importing mechanism will be now deprecated.

the following warning will be issued:

    The python-import-mechanism used for application "' + application_name
    is DEPRECATED!
    Please check the following website for instuctions on how to update it:
    https://github.com/KratosMultiphysics/Kratos/wiki/Applications-as-python-modules
    The old mechanism will be removed on 01.10.2019!

important messages:

* current code will stop working on 1/10/2019
* info can be found in the wiki page [DOC](https://github.com/KratosMultiphysics/Kratos/wiki/Applications-as-python-modules)

**Comment by philbucher** [comment]
@KratosMultiphysics/all

**Comment by jcotela** [comment]
Will the core KratosMultiphysics module also be updated to a pythonic module structure on the mentioned date?

**Comment by philbucher** [comment]
I would say so, then we only have the change once

Ofc this still requires some updates in the Core itself but we can do it on the way

BTW we are talking abt taking out this line:
https://github.com/KratosMultiphysics/Kratos/blob/dc0d9e31bec2b685e534848dad108d5fe75621be/kratos/python_interface/__init__.py#L20-L22

**Comment by pooyan-dadvand** [comment]
> BTW we are talking abt taking out this line:

Then it will stop working?

**Comment by philbucher** [comment]
It will only work with the „full“ path, i.e.
from KratosMultiphysics import xxx

Only doing
import xxx
Will stop working 

=> will be the same behavior as for the applications

**Comment by RiccardoRossi** [comment]
how is the current status of this?

**Comment by philbucher** [comment]
pretty good

Travis compiles with #5460

**Comment by maceligueta** [comment]
The DEM and Swimming DEM were ported and the warning is no longer appearing.

**Comment by rubenzorrilla** [comment]
Just tagging all since there is one month left to apply the changes. @KratosMultiphysics/all

**Comment by ipouplana** [comment]
Poromechanics, FluidTransport and PfemFluid applications are already updated.

**Comment by AlejandroCornejo** [comment]
FemToDemApplication already updated, Thank you.

**Comment by joaquinirazabal** [comment]
The DamApplication is also updated.

**Comment by RiccardoRossi** [comment]
i think we need to be aware that python doesn t allow to do

~~~
import KratosMultiphysics.FluidDynamicsApplication as cfd
import cfd.specific_python_script
~~~

that is, we cannot import from an alias but rather need the full name of tje package.

see
https://stackoverflow.com/questions/42459939/import-modules-using-an-alias?rq=1

for discussion (it also surprised me btw)

**Comment by philbucher** [comment]
Thats what I meant last time we discussed it
Alternatively we could import the py-scripts in the init, then this would work I am quite sure 
-> thats how we do it with the C-things

**Comment by RiccardoRossi** [comment]
> Thats what I meant last time we discussed it

true, but as usually i didn't believe it until trying ;-)

> Alternatively we could import the py-scripts in the init, then this would work I am quite sure
> -> thats how we do it with the C-things

i have contrasted feelings about this: on one side this way python-only stuff would behave as c++ stuff, on the other side this implies that once you import an application you import ALL of it (so a lot of potential clashes)

**Comment by philbucher** [comment]
:)

Thats the reason @jcotela and I didn’t try to tight away push into this direction since we knew that there are several unanswered things to be discussed
=> now is the time and I agree that this is very useful/ can shorten our imports

**Comment by jcotela** [comment]
@RiccardoRossi if all names are imported through relative imports (which we are already enforcing), name clashes should be relatively harder. For me, the main drawback is that all the python things we want to make visible automatically should be "registered" by importing them from the application init, which seems like a maintenance chore and a source of conflicts.

**Comment by RiccardoRossi** [comment]
can't we have a "*" import in the __init__ which is located within python_scripts ?

**Comment by jcotela** [comment]
does something like `from . import *` work? I've never tried, but my understanding was that you still need to give the file name.

**Comment by philbucher** [comment]
I will give it a try later

**Comment by philbucher** [comment]
so I played around a bit, here are my findings
- doing `from . import *` inside of `KratosMultiphysics/XXXApplication/__init__.py` does nothing
- same for `from KratosMultiphysics.XXXApplication import *`
- doing `from . import py_script` inside of `KratosMultiphysics/XXXApplication/__init__.py` works **IF** there is no import in the style `import KratosMultiphysics.XXXApplication as xxxApp` (gives: `AttributeError: module 'KratosMultiphysics' has no attribute 'XXXApplication'`) 
    => `from KratosMultiphysics import XXXApplication as xxxApp` works though, no clue why
- I also tried looping the files in the `python_scripts` directory (basically to achieve `from . import *`) and loading them with `importlib` but that does also not work => `ImportError: No module named KratosMultiphysics.XXXApplication.py_script.py, KratosMultiphysics.XXXApplication.py_script.py is not a package`
- **EDIT**: I also tried with a script inside of `python_scripts` (`import_all.py`) which contained `from . import *`. I imported this in the `KratosMultiphysics/XXXApplication/__init__.py` but it also didn't do anything

Independent of what seems to be technically possible I am still not convinced we should go into the `from . import *` direction to also import the py-scripts. Reason is that many files have dependencies (e.g. the python-trilinos solvers) that will crash then. 
Also then we by default import everything, even though it might not be used. I would rather prefer to keep the dependencies to a minimum and only import the things we actually use (at least on the py-level).
